sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -188,11 +203,61 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     }
   }
 
+  // Here is a custom layer that allows us to abort requests that go through MSAL,
+  // since MSAL doesn't allow us to pass options all the way through.
+
+  generateAbortSignal(correlationId?: string): AbortSignalLike {
+    const controller = new AbortController();
+    const key = correlationId || "noCorrelationId";
+
+    const controllers = this.abortControllers.get(key) || [];
+    controllers.push(controller);
+    this.abortControllers.set(key, controllers);
+
+    return controller.signal;
+  }
+
+  abortRequests(correlationId?: string): void {
+    const key = correlationId || "noCorrelationId";
+    const controllers = [
+      ...(this.abortControllers.get(key) || []),
+      // MSAL passes no correlation ID to the get requests...
+      ...(this.abortControllers.get("noCorrelationId") || [])
+    ];
+    if (!controllers.length) {
+      return;
+    }
+    for (const controller of controllers) {
+      controller.abort();
+    }
+    this.abortControllers.set(key, undefined);
```

Comment:
since we always grab the "noCorrelationId" ones, shouldn't we always clear them here too?

Reply:
sounds good! thank you!